src/components/JourneyComparison/FrequencyComparison.tsx

```diff
@@ -17,22 +17,49 @@ interface FrequencyInfoProps {
 }
 
 const getStationIdToCompare = (baseline: JourneyInfo, enhanced: JourneyInfo) => {
-    const [baselineFirstSegment, enhancedFirstSegment] = [baseline, enhanced].map(
-        (journey) => journey.segments.find((seg) => seg.kind === "travel") as JourneyTravelSegment
-    );
-    const hasArrivals =
-        baseline.arrivals[baselineFirstSegment.startStation.id] &&
-        enhanced.arrivals[enhancedFirstSegment.startStation.id];
-    const eitherJourneyStartsWithRegionalRail =
-        isRegionalRailRouteId(baselineFirstSegment.routeId) ||
-        isRegionalRailRouteId(enhancedFirstSegment.routeId);
-    return hasArrivals && eitherJourneyStartsWithRegionalRail
-        ? baselineFirstSegment.startStation.id
+    let firstSegment;
```

Comment:
My proposal is to make this function easier to read by:

1. Creating a `getFirstStationAndRouteFromJourney` helper function that tries to extract the `startStation.id` and `routeId` from a journey's first travel segment, and returning `null` for each if it fails (i.e. the `JourneyInfo.segments` array is empty).
2. Removing the `hasArrivals` check entirely, since I think you'll have to be on guard for that in downstream code anyway.
3. Returning `eitherJourneyStartsWithRegionalRail ? (baselineStartStationId || enhancedStartStationId) : null`

I think this reduces the function to a few lines:

```ts
const getStationIdToCompare = (baseline: JourneyInfo, enhanced: JourneyInfo) => {
    const [
        { routeId: baselineRouteId, stationId: baselineStationId },
        { routeId: enhancedRouteId, stationId: enhancedStationId },
    ] = [baseline, enhanced].map(getFirstStationAndRouteFromJourney);

    const eitherJourneyStartsWithRegionalRail =
        (baselineRouteId && isRegionalRailRouteId(baselineRouteId)) ||
        (enhancedRouteId && isRegionalRailRouteId(enhancedRouteId));

    return eitherJourneyStartsWithRegionalRail ? baselineStationId || enhancedStationId : null;
};
```
and removes any explicit error checking altogether, expressing the "navigation failed" state in terms of null values from `getFirstStationAndRouteFromJourney`.